src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtil.java

```diff
@@ -437,6 +437,28 @@ public static boolean isReceiverParameter(DetailAST parameterDefAst) {
                 && parameterDefAst.findFirstToken(TokenTypes.IDENT) == null;
     }
 
+    /**
+     * Returns the access modifier of the method/constructor at the specified AST. If
+     * the method is in an interface or annotation block, the access modifier is assumed
+     * to be public.
+     *
+     * @param ast the token of the method/constructor.
+     * @return the access modifier of the method/constructor.
+     */
+    public static AccessModifierOption getAccessModifierFromModifiersToken(DetailAST ast) {
+        final AccessModifierOption accessModifier;
+
+        if (ScopeUtil.isInInterfaceOrAnnotationBlock(ast)) {
+            accessModifier = AccessModifierOption.PUBLIC;
```

Comment:
Since Java9 interface methods can [be private](https://bugs.openjdk.java.net/browse/JDK-8071453). Please change this to the __default__ value of `accessModifier` if no access modifier is explicitly set.

Reply:
Changing this to `AccessModifierOption.PACKAGE` causes changes in existing tests for `ParameterNameCheck`, since this code was basically extracted from that check. I think we should make this a separate issue.